flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/FlinkKafkaInternalProducer.java

```diff
@@ -154,20 +161,6 @@ public void close() {
                 "Close without timeout is now allowed because it can leave lingering Kafka threads.");
     }
 
-    @Override
-    public void close(long timeout, TimeUnit unit) {
-        synchronized (producerClosingLock) {
-            kafkaProducer.close(timeout, unit);
-            if (LOG.isDebugEnabled()) {
-                LOG.debug(
-                        "Closed internal KafkaProducer {}. Stacktrace: {}",
-                        System.identityHashCode(this),
-                        Joiner.on("\n").join(Thread.currentThread().getStackTrace()));
-            }
-            closed = true;
-        }
-    }
-
```

Comment:
Don't we need this anymore or is this no longer possible?

Reply:
The method is no longer part of the interface therefore we cannot implement it anymore.

Reply:
Was it dead code before? (looks to me like it)

Reply:
Why should be it dead code?

Reply:
Ah sorry, I looked at the wrong commit.

Reply:
Kafka didn't remove this method in 2.8.1, it removed it in 3.0.0
https://github.com/apache/kafka/blob/7
2.8/clients/src/main/java/org/apache/kafka/clients/producer/Producer.java#L104

Reply:
Thanks for the hint during the upgrade test I also tested with 3.0.0. Probably I missed that when going back to 2.8.